### ISSUE TITLE
Behavior change when "extra_config" > "router" is populated in krakend.json. X-Forwarded-For and ips field are not what's expected

### ISSUE THREAD

**Issue by Paraplegix** [krakendio/krakend-ce#689]
**Environment info:**

* KrakenD version: tested with docker 2.0.4, 2.2.0 (screenshots) and latest (2.2.1) as of writing the bug report. Bug seen on kubernetes hosted on GCP
* System info: Darwin 22.2.0 arm64
* Hardware specs: M1 8Go
* Backend technology: All of them ?
* Additional environment information:

**Describe the bug**
It seems that when there is anything inside the field "router" of "extra_config", the behavior of the router change. I do not know to what extend it changes, but at least the handling of the header X-Forwarded-For is impacted

I would expect that when we receive a header X-Forwarded-For, its content should be enriched with krakend Ip and forwarded to the pod or at least the first IP in the list should be used as the origin Ip.
Currently without any configuration in the router field, the first IP in the list is used as the origin Ip wich is fine for our usecase.
But with any configuration in the router field, a local ip is used instead of the one in X-Forwarded-For. 

We are seeing this on our production servers. Krakend is deployed on a kubernetes cluster and is behind nginx and google http balancer [that uses X-Forwarded-For to forward callers Ip](https://cloud.google.com/load-balancing/docs/https)
We also receive request from other servers that are forwarded for an end user. Those server use X-Forwarded-For transfering the user true ip.

**Your configuration file**:
I've recreated a very basic configuration file for error reproduction, a single endpoint that redirect on a running local pod using [
mendhak/http-https-echo](https://github.com/mendhak/docker-http-https-echo) to view what the backend is receiving

For testing change what is inside "extra_config" > "router"

As is, this configuration provoke the bug. You can remove "hide_version_header" (changing to false is not enough) to have what I think is an acceptable behavior.
```json
{
  "$schema": "https://www.krakend.io/schema/v3.json",
  "version": 3,
  "name": "KrakenD - API Gateway",
  "timeout": "3000ms",
  "cache_ttl": "300s",
  "output_encoding": "json",
  "extra_config": {
    "router": {
      "hide_version_header":true
    }
  },
  "endpoints": [
    {
      "endpoint": "/echo",
      "method": "GET",
      "output_encoding": "json",
      "input_headers": ["*"],
      "backend": [
        {
          "url_pattern": "/",
          "encoding": "json",
          "sd": "static",
          "method": "GET",
          "host": [
            "host.docker.internal:9080"
          ],
          "disable_host_sanitize": false
        }
      ]
    }
  ]
}
```

**Commands used**
How did you start the software?
```
docker run -p 8080:8080 --rm -t -v $PWD:/etc/krakend/ devopsfaith/krakend:2.2.0 run --config /etc/krakend/krakend.json
```

**Expected behavior**
Unless there are configuration that explicitly change how X-Forwarded-For should be handled, the source ip should be the one transfered to my pod and not the gateway ip.

For example, when receiving on krakend a request with X-Forwarded-For containing "A.A.A.A, B.B.B.B, C.C.C.C" I would expect krakend to either forward the header as is with more information (Krakend append the caller ips to the header) or at least keep the first Ip "A.A.A.A" in this case.

**Additional context**
Some screenshot of example of reproduction. General layout is on the left the request sent with header and response, and on the right the krakend json content + logs from krakend started as a docker pod.

1. [With just extra_config (no router)](https://imgur.com/7YPZr3w), the request is forwarded with the correct ip
1. [With extra_config > router but empty config](https://imgur.com/hRpKhUp), the request is forwarded with the correct ip
1. [With extra_config > router > "hide_version_header": true](https://imgur.com/ujR9vym) , the request is forwarded with the gateway ip (172.17.0.1)
1. [With extra_config > router > "hide_version_header": false](https://imgur.com/z0vjkdd) (false being default value), the request is forwarded with the gateway ip (172.17.0.1)
1. [With extra_config > router  with a non existing key ("potatoes":"fries")](https://imgur.com/WEUUZwf), the request is forwarded with the gateway ip  (172.17.0.1). This screenshot include test for version 2.0.4, 2.2.0 and 2.2.1

**Comment by alombarte** [comment]
Hi @Paraplegix , thanks for the detailed issue. I wish all issues were like yours.

Thanks to your details we were able to quickly identify the problem, and the bug will be fixed in Lura.

**Comment by jandelgado** [comment]
Krakend version v2.3.2 uses lura v2.2.7, which has the mentioned fix (luraproject/lura#654). However in my tests the problem still shows up.  If needed I can provide docker-compose.yaml and curl examples to reproduce. @alombarte could you pleas check if the fix is effective? thanks
